examples/nametoken/src/lib.rs

```diff
@@ -0,0 +1,1849 @@
+//! A NameToken smart contract example using the Concordium Token Standard
+//! CIS2.
+//!
+//! # Description
+//! A contract that allows for registering and managing names. A name can be
+//! used as a verifiable human-readable name for an address. They also can have
+//! some data attached, like website URLs, accounts in various systems, etc.
+//! Only account addresses can be owners. The ownership can be transferred to
+//! another account address.
+//!
+//! # Operations
+//! The contract allows for
+//!  - registering new names;
+//!  - updating data associated with a name;
+//!  - renewing a name;
+//!  - viewing information about a name (owner and data).
+//!
+//! To register, renew and update one has to pay a fee. Registration succeeds if
+//! either name is not yet registered, or it is registered but expired. In this
+//! case, ownership is transferred to the new owner and the expiration date is
+//! updated as for the "fresh" registration. Updating data and renewing is only
+//! possible if the name is not expired.
+//!
+//! # Token
+//! NameToken is essentially an NFT that adheres to the CIS-2 token standard.
+//! This allows for trading of domain names on NFT auctions. The contract is
+//! based on the NFT example with modifications required for the name
+//! management. Registering a fresh name is effectively minting. It generates
+//! log events accordingly. However, taking over an expired name is not
+//! considered minting. Ownership can be transferred by the owner or by an
+//! operator. Owners are accounts only, but operators can be any address. Token
+//! ownership is determined by the expiration date. Expired name tokens are not
+//! burned, they are considered as not owned, that is the balance is 0 for the
+//! address it was initially registered.
+//!
+//! Note: token ids are hashed names. The words "name" and "token id" are used
+//! interchangeably.
+//!
+//! # Misc
+//! This example demonstrates how to use crypto primitives (hashing) and
+//! lazy-loaded data.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use concordium_cis2::*;
+use concordium_std::*;
+
+/// The baseurl for the token metadata, gets appended with the token ID as hex
+/// encoding before emitted in the TokenMetadata event.
+const TOKEN_METADATA_BASE_URL: &str = "https://some.example/nametoken/";
+
+/// List of supported standards by this contract address.
+const SUPPORTS_STANDARDS: [StandardIdentifier<'static>; 2] =
+    [CIS0_STANDARD_IDENTIFIER, CIS2_STANDARD_IDENTIFIER];
+
+// Fees
+
+// Registration fee in CCD
+const REGISTRACTION_FEE: Amount = Amount::from_ccd(70);
+
+// Data update fee in CCD
+const UPDATE_FEE: Amount = Amount::from_ccd(7);
+
+// Renewal fee in CCD
+const RENEWAL_FEE: Amount = Amount::from_ccd(7);
+
+// How long the registered name is owned before it needs to be renewed
+const REGISTRATION_PERIOD_DAYS: u64 = 365;
```

Comment:
@limemloh do you mean to make this part of the state and let the users initialise it in `init`?

Reply:
No, I mean changing this line to something like:
```suggestion
const REGISTRATION_PERIOD_DAYS: Duration = Duration::from_days(365);
```
But this will currently result in a compile error, since we did not mark `Duration::from_days` as a [`const`-function](https://doc.rust-lang.org/std/keyword.const.html#compile-time-evaluable-functions).
So you would have to make a another PR adding the `const` mark before doing this

Reply:
Ah, ok. I'll leave it as it is for now here and make a PR to `concordium-contracts-common`. We can update the example later on.